sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationRecordedTestSanitizer.cs

```diff
@@ -1,21 +1,21 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using Azure.Core;
 using Azure.Core.Testing;
 
 namespace Azure.Data.AppConfiguration.Tests
 {
     public class ConfigurationRecordedTestSanitizer : RecordedTestSanitizer
     {
-        public override void SanitizeConnectionString(ConnectionString connectionString)
+        public override string SanitizeConnectionString(string connectionString)
```

Comment:
What's the reason for this change?

Idea of this methods signature was that you can call base with the same instance of `ConnectionString` and combine the effect. Now every implementor needs to parse/ToString.

Reply:
`ConnectionString` is an internal type, while `RecordedTestSanitizer.SanitizeConnectionString` is public.

Reply:
I'm a little confused here.  @AlexanderSher it looks like you've introduced a new internal class ConnectionString under core/src/Shared, whereas there was already a public ConnectionString type under core/tests/TestFramework.  Would it be possible to unify these two classes?  It's confusing to have two classes with the same name in the same project, and we should have a good reason to do this if we do.

I also want to make sure we address Pavel's comment -- it sounds like changing this method signature violates the intention behind how the method was designed.  Can we find a way to achieve these goals without going against the intended design?  If needed, let's work with Pavel on this.

Reply:
> Would it be possible to unify these two classes?

That's exactly what I've done (sorry, I've forgotten to delete `sdk/core/Azure.Core/tests/TestFramework/ConnectionString.cs` in original commit, fixed now).

> it sounds like changing this method signature violates the intention behind the method's design.

To preserve original design, we either need to make `sdk/core/Azure.Core/src/Shared/ConnectionString.cs` public or convert `TestFramework/ConnectionString` into public wrapper and use it in tests. Up to @pakrym .

Reply:
I see, what you currently have is fine.